File: src/de/saschahlusiak/freebloks/view/effects/AbsEffect.java
package de.saschahlusiak.freebloks.view.effects;


public abstract class AbsEffect implements Effect {
	AbsEffect() {
		time = 0.0f;
	}

	float time;

	public boolean execute(float elapsed) {
		time += elapsed;
		return true;
	}

	public boolean isEffected(int x, int y) {
		return false;
	}
}

File: src/main/java/de/saschahlusiak/freebloks/view/effects/AbsEffect.kt
package de.saschahlusiak.freebloks.view.effects

import de.saschahlusiak.freebloks.view.BoardRenderer
import javax.microedition.khronos.opengles.GL11

abstract class AbsEffect internal constructor() : Effect {
	internal var time = 0.0f

    override fun execute(elapsed: Float): Boolean {
        time += elapsed
        return true
    }

    override fun isEffected(x: Int, y: Int): Boolean {
        return false
    }

    override fun renderShadow(gl: GL11, renderer: BoardRenderer) { }
}